src/main/java/seedu/module/ui/ModuleListPanel.java

```diff
@@ -38,7 +39,7 @@ protected void updateItem(Module trackedModule, boolean empty) {
                 setGraphic(null);
                 setText(null);
             } else {
-                setGraphic(new ModuleCard(trackedModule, getIndex() + 1).getRoot());
+                setGraphic(new ModuleCard((ModuleTracked) trackedModule, getIndex() + 1).getRoot());
```

Comment:
This is not safe. I realised that the Ui could display either the ArchivedModule or TrackedModule, so it will break when it tries to display ArchivedModule this way. Refer to the above comment on separating the interfaces. Maybe we can typecheck for `TrackedModule` and display extra info accordingly?